### PR TITLE
Files: use app-layout and sidebar tabs

### PR DESCRIPTION
**Pull request by tomneedham**
Use app layout in files app.
Ripped out old vuetify files version and files info sidebars.
Fixed loading of sidebars to ensure vuex state is propagated.
Use oc-tabs and oc-tab-items for registered file sidebars.

Do we need `@reload="$emit('reload')` in the dynamic component?

### CODE REVIEW

File: apps/files/src/components/FileDetails.vue [hunk 1]
```diff
@@ -1,29 +1,14 @@
...
-          :key="tab.name"
-        >
-          <component :is="tab.component" @reload="$emit('reload')"></component>
-        </v-tab-item>
-      </v-tabs>
+      <oc-tabs>
+          <oc-tab-item v-for="tab of fileSideBars" :title="tab.name" :key="tab.name" />
+      </oc-tabs>
+      <p>hello! {{ fileSideBars.length }}</p>
+      <component :is="fileSideBars[active]" @reload="$emit('reload')"></component>
```
Comment (DeepDiver1975) [inline apps/files/src/components/FileDetails.vue]: put thin into a computed value - no idea if binding works on the array accessor

File: apps/files/src/components/FileDetails.vue [hunk 2]
```diff
@@ -32,29 +16,30 @@
...
   },
   methods: {
+    log (index) {
+      alert(index)
+    },
     ...mapActions('Files', ['deleteFiles']),
     close () {
       this.$emit('reset')
     },
-    showSidebar (sideBarName) {
```
Comment (DeepDiver1975) [inline apps/files/src/components/FileDetails.vue]: Please keep this. It is used to highlight one Tab when hitting an Action button in the filelist

File: apps/files/src/components/FileDetails.vue [hunk 3]
```diff
@@ -32,29 +16,30 @@
...
+      activeTab: 0
     }
   },
   components: {
-    OcAppSideBar
+    OcAppSideBar,
+    FileSharingSidebar
   },
   methods: {
+    log (index) {
```
Comment (tomneedham) [inline apps/files/src/components/FileDetails.vue]: 💣

File: apps/markdown-editor/src/app.js [hunk 4]
```diff
@@ -31,6 +31,13 @@ const appInfo = {
     extension: 'md'
     // icon: 'custom_icon_class_to_override_icon_for_filetype_md'
   }
+  ],
+  fileSideBars: [
```
Comment (DeepDiver1975) [inline apps/markdown-editor/src/app.js]: remove this one .... there is no markdown sidebar ;-)

File: src/components/Menu.vue [hunk 5]
```diff
@@ -34,7 +34,7 @@ export default {
       }
     },
     _logoutItemText () {
-      return this.$gettextInterpolate(this.$gettext('Exit %{product}'), { product: this.configuration.theme.general.name })
```
Comment (DeepDiver1975) [inline src/components/Menu.vue]: why hard code the name again?

File: src/components/OcAppSideBar.vue [hunk 6]
```diff
@@ -2,14 +2,14 @@
...
-            <span>Heading</span>
-          </slot>
-        </div>
         <div v-if="!disableAction" class="action">
           <slot name="action">
-            <button @click="$emit('close', $event)"><v-icon color="white" large>close</v-icon></button>
+            <oc-button icon="close" class="uk-float-right uk-margin-small-right" @click="$emit('close', $event)"></oc-button>
+          </slot>
+        </div>
+        <div class="sidebar-title">
```
Comment (DeepDiver1975) [inline src/components/OcAppSideBar.vue]: I don't think we need this component at all - should come from ods - right?

Comment (butonic) [reply src/components/OcAppSideBar.vue]: in a subsequent PR

**Pull request merged**